diff --git a/dist/runtime/types/tv.d.ts b/dist/runtime/types/tv.d.ts
index 55ef2844ce38ca060c85ad9b22e1c0a5c900a23f..91919f03c42350542f3469d89a278dcc4668565a 100644
--- a/dist/runtime/types/tv.d.ts
+++ b/dist/runtime/types/tv.d.ts
@@ -1,4 +1,4 @@
-import type { ClassValue, TVVariants, TVCompoundVariants, TVDefaultVariants } from 'tailwind-variants';
+import type { ClassValue, StringToBoolean, TVVariants, TVCompoundVariants, TVDefaultVariants } from 'tailwind-variants';
 /**
  * Defines the AppConfig object based on the tailwind-variants configuration.
  */
@@ -26,7 +26,7 @@ type Id<T> = {} & {
 type ComponentVariants<T extends {
     variants?: Record<string, Record<string, any>>;
 }> = {
-    [K in keyof T['variants']]: keyof T['variants'][K];
+    [K in keyof T['variants']]: StringToBoolean<keyof T['variants'][K]> | (string & {});
 };
 type ComponentSlots<T extends {
     slots?: Record<string, any>;
